### DISCUSSION TITLE
PostgreSQL Usage

### DISCUSSION THREAD

**Discussion by emert117** [yanpitangui/dotnet-api-boilerplate#88] [General]
Hi, I want to use PostgreSQL database instead of MS SQL Server. I will use a managed database hosting service so I don't want to use docker image of PostgreSQL. I want to add only the connection string of PostgreSQL server.

What is the best way to do this?

**Comment by yanpitangui** [comment]
Hello!
First, you have to install in your infrastructure project the Ef provider for postgres package: https://www.nuget.org/packages/Npgsql.EntityFrameworkCore.PostgreSQL
Then, you have to replace the "DefaultConnection" present in the appsettings with your hosted pg instance connection string (or use [user-secrets](https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-7.0&tabs=windows).
In the file PersistanceSetup.cs, replace the call to UseSqlServer to UseNpgsql.
That's it. You will probably have to remake your migrations, as I think they are specific to MS SQL Server.

**Comment by emert117** [comment]
Thanks
